### DISCUSSION TITLE
Relational O2O data in JSON import file

### DISCUSSION THREAD

**Discussion by tomniemi** [directus/directus#15399] [Archive]
Last week I managed to get relational O2M data import working by looking at the Directus Export result. Discussion can be found here: [https://github.com/directus/directus/discussions/15307](https://github.com/directus/directus/discussions/15307)

So for example the following O2M structure is working:

`[    
    {
        "id": 1,
        "test": "TEST 1",
        "test_child": 
        [
            {
                "id": 2,
                "test_parent_id": 1,
                "test": "CHILDTEST 1"
            }
        ]
    }
]`

Now I am trying to import O2O data like this:

`[    
    {
        "id": 1,
        "test": "TEST 1",
        "test_child":         
         {
             "id": 2,
             "test_parent_id": 1,
             "test": "CHILDTEST 1"
         }        
    }
]`

I followed the guidelines in Directus Docs to set up O2O relationship: [https://docs.directus.io/configuration/data-model/relationships.html#one-to-one-o2o](https://docs.directus.io/configuration/data-model/relationships.html#one-to-one-o2o). Especially I configured field's schema and toggled 'Unique' option on (this creates new unique constraint to database). 

Despite of configurations, both Export and Import only works with O2M structure (first code sample). Is it possible to get O2O working (second code sample)?

**Comment by br41nslug** [comment]
> Directus does not include a dedicated One-to-One (O2O) relationship type or Interface. 

As there is not a true O2O type the first example is the correct format because an O2O relation will be a M2O relation with 1 item in it.

**Reply by tomniemi** [reply]
@br41nslug thank you for your answer!

I think in many Import/Export cases it would be useful to have the ability to configurate O2O relationship (or use the unique database constraint as an indicator for O2O relationship to format Import/Export outcome). 

For example if you have an existing external data sources with O2O relationships, it might be very difficult to change the external data source (many other systems may rely on the output of that external system so changes to that JSON would affect many other systems).
